core/controllers/learner_dashboard.py

```diff
@@ -0,0 +1,206 @@
+                'category': collection_summary.category,
+                'objective': collection_summary.objective,
+                'language_code': collection_summary.language_code,
+                'last_updated': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_last_updated),
+                'created_on': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_created_on),
+                'status': collection_summary.status,
+                'node_count': collection_summary.node_count,
+                'community_owned': collection_summary.community_owned,
+                'thumbnail_icon_url': (
+                    utils.get_thumbnail_icon_url_for_category(
+                        collection_summary.category)),
+                'thumbnail_bg_color': utils.get_hex_color_for_category(
+                    collection_summary.category),
+            })
+
+        for collection_summary in incomplete_collection_summaries:
+            incomplete_collection_summary_dicts.append({
+                'id': collection_summary.id,
+                'title': collection_summary.title,
+                'category': collection_summary.category,
+                'objective': collection_summary.objective,
+                'language_code': collection_summary.language_code,
+                'last_updated': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_last_updated),
+                'created_on': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_created_on),
+                'status': collection_summary.status,
+                'node_count': collection_summary.node_count,
+                'community_owned': collection_summary.community_owned,
+                'thumbnail_icon_url': (
+                    utils.get_thumbnail_icon_url_for_category(
+                        collection_summary.category)),
+                'thumbnail_bg_color': utils.get_hex_color_for_category(
+                    collection_summary.category),
+            })
+
+        creators_subscribed_to = (
+            subscription_services.get_all_creators_subscribed_to(self.user_id))
+        creators_settings = user_services.get_users_settings(
+            creators_subscribed_to)
+        subscription_list = []
+
+        for index, creator_settings in enumerate(creators_settings):
+            subscription_summary = {
+                'creator_picture_data_url': (
+                    creator_settings.profile_picture_data_url),
+                'creator_username': creator_settings.username,
+                'creator_impact': (
+                    user_services.get_user_impact_score(
+                        creators_subscribed_to[index]))
+            }
+
+            subscription_list.append(subscription_summary)
+
+        self.values.update({
+            'completed_explorations_list': completed_exp_summary_dicts,
```

Comment:
Btw, note that you might need to test this with several explorations to see the effects in practice. It might be worth collaborating with @shaz13 and @souravbadami on #2508 if there are utility functions in the admin dashboard for this that you'd all find useful. In the meanwhile, though, it should be possible to batch the datastore calls.